### PR TITLE
Admin pages tweaks

### PR DESCRIPTION
**Pull request by marcospri**
For both registrations and application instances:

- Search by ID.
- Searching with no criteria returns some results.
- Don't redirect to the result when there only one. It's a bit confusing, you just might want to copy something from the results table for example.
- Correctly 404 when registration doesn't exist.

## Testing

- http://localhost:8001/admin/registrations/

  - Search directly there, you'll get results
  - Search using ID `1`
  - http://localhost:8001/admin/registration/id/78789789/ is a 404 not an empty page

- http://localhost:8001/admin/instances/

  - Search directly there, you'll get results
  - Search using ID `100`

### CODE REVIEW

File: tests/unit/lms/services/application_instance_test.py [hunk 1]
```diff
@@ -63,7 +63,7 @@ def test_get_by_id(self, service, application_instance):
 
     def test_get_by_not_found(self, service):
         with pytest.raises(ApplicationInstanceNotFound):
-            service.get_by_id(0)
+            service.get_by_id(100_000_000)
```
Comment (marcospri) [inline tests/unit/lms/services/application_instance_test.py:66]: `0` not a valid ID, use a real one.

File: tests/unit/lms/services/lti_registration_test.py [hunk 2]
```diff
@@ -57,6 +61,7 @@ def test_search_registrations(self, svc):
 
     @pytest.fixture
     def registration(self):
+        factories.LTIRegistration.create_batch(4)
```
Comment (marcospri) [inline tests/unit/lms/services/lti_registration_test.py:64]: Add some more regs here.

**Comment by jon-betts** [comment]
Testing results:

 * Visit: http://localhost:8001/admin/registrations
 * Enter 1
 * Search works and shows a value at the bottom
 * Enter 87465873465
 * :lady_beetle:  **Search crashes with "Sorry, but something went wrong"** !
 * Visit: http://localhost:8001/admin/registration/id/78789789/
 * Shows a plain 404
 
 * Visit: http://localhost:8001/admin/instances
 * Enter 100
 * Search works and shows a value at the bottom
 * Enter 87465873465
 * :lady_beetle:  **Search crashes with "Sorry, but something went wrong"** !
 * Visit: http://localhost:8001/admin/instance/id/23453454/
 * Shows a plain 404

**Comment by marcospri** [comment]
> ## Testing results
> Registrations
> 
> * Visit: http://localhost:8001/admin/registrations
> * Enter 1
> * Search works and shows a value at the bottom
> * Enter 87465873465
> * lady_beetle  **Search crashes with "Sorry, but something went wrong"** !


Fixed this in a new commit for registrations, application instances and organizations (which already had the search by ID).

File: lms/views/admin/__init__.py [hunk 3]
```diff
@@ -1,10 +1,34 @@
+from marshmallow import fields, missing
 from pyramid.httpexceptions import HTTPFound, HTTPNotFound
 from pyramid.renderers import render_to_response
 from pyramid.view import forbidden_view_config, notfound_view_config, view_config
 
 from lms.validation._exceptions import ValidationError
 
 
+class EmptyStringNoneMixin:
```
Comment (marcospri) [inline lms/views/admin/__init__.py:9]: This is very verbose but I reckon we should handle any of this validations with Marshmallow.

File: lms/views/admin/lti_registration.py [hunk 4]
```diff
@@ -240,3 +235,9 @@ def _get_url_suggestions(issuer, client_id):
                 }
 
         return {"auth_login_url": None, "key_set_url": None, "token_url": None}
+
+    def _get_registration_or_404(self, id_) -> LTIRegistration:
+        if org := self.lti_registration_service.get_by_id(id_):
+            return org
+
+        raise HTTPNotFound()
```
Comment (jon-betts) [inline lms/views/admin/lti_registration.py:243]: This isn't very fancy, but users shouldn't be typing in URLs by hand, so this seems fine.

**Pull request merged**